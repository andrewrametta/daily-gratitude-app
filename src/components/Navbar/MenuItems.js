export const MenuItems = [
  {
    title: "Home",
    url: "/",
    cName: "nav-links",
  },
  {
    title: "About",
    url: "/about",
    cName: "nav-links",
  },
  {
    title: "Login",
    url: "/login",
    cName: "nav-links",
  },
  {
    title: "Register",
    url: "/register",
    cName: "nav-links",
  },
  {
    title: "User",
    url: "/user",
    cName: "nav-links",
  },
  {
    title: "New Day",
    url: "/new-day",
    cName: "nav-links",
  },
];
