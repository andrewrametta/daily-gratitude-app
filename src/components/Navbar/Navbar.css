a {
  color: #122663;
  text-decoration: none;
}
a:hover {
  color: #177876;
}

.logo-image {
  max-width: 240px;
}

.hamburger {
  list-style: none;
  padding: 0;
  flex-direction: column;
  justify-content: space-between;
  display: none;
}
.line {
  width: 20px;
  height: 2px;
  background-color: black;
}

.navbar {
  display: flex;
  position: relative;
}
.logo {
  flex: 1;
  margin: 0;
}
.nav {
  display: flex;
  list-style: none;
  padding: 0;
  align-items: center;
}
.nav-item {
  margin: 0 10px;
}

@media screen and (max-width: 600px) {
  .logo-image {
    max-width: 180px;
  }

  .nav {
    background-color: white;
    position: absolute;
    flex-direction: column;
    left: 0;
    right: 0;
    top: 20px;
    display: none;
    padding: 20px;
  }
  .hamburger {
    display: flex;
    height: 20px;
    padding-right: 10px;
  }
  .hamburger:hover + .nav,
  .nav:hover {
    display: block;
  }
  .nav-item {
    width: 100%;
    text-align: center;
    padding: 20px 0;
  }
  .nav-item:hover {
    background-color: rgb(#177876);
  }
}
