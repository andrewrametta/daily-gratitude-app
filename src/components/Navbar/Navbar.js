import React, { Component } from "react";
import { Link, NavLink } from "react-router-dom";
import TokenService from "../../services/token-service";
import Context from "../../Context";
import "./Navbar.css";

class Navbar extends Component {
  static contextType = Context;

  logout = () => {
    TokenService.clearAuthToken();
    this.props.history.push("/");
  };

  render() {
    return (
      <nav className="navbar">
        <Link className="logo" to="/">
          <img className="logo-image" src="/logo.svg" alt="Daily Gratitude" />
        </Link>
        <ul className="hamburger">
          <li className="line"></li>
          <li className="line"></li>
          <li className="line"></li>
        </ul>
        <ul className="nav">
          {TokenService.hasAuthToken() ? (
            <>
              <li className="nav-item">
                <Link to="/dashboard">Dashboard</Link>
              </li>
              <li className="nav-item">
                <NavLink to="/new-day">Add New Entry</NavLink>
              </li>
              <li className="nav-item">
                <NavLink to="/" onClick={this.logout}>
                  Logout
                </NavLink>
              </li>
            </>
          ) : (
            <>
              <li className="nav-item">
                <NavLink to="/register">Register</NavLink>
              </li>
              <li className="nav-item">
                <NavLink to="/login">Login</NavLink>
              </li>
            </>
          )}
        </ul>
      </nav>
    );
  }
}

export default Navbar;
