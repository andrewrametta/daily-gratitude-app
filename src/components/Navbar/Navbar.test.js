import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Navbar from "./Navbar";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe(`Navbar component`, () => {
  it("renders the complete page", () => {
    const wrapper = shallow(<Navbar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
