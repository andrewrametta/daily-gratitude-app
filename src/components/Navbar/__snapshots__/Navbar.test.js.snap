// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders the complete page 1`] = `
<nav
  className="NavbarItems"
>
  <div
    className="meun-icon"
  />
  <ul
    className="ul-nav"
  >
    <li
      className="li-nav-logo"
    >
      <Link
        className="logo"
        to="/"
      >
        Daily Gratitude
      </Link>
    </li>
    <li
      className="li-nav"
    >
      <NavLink
        className="navLinks"
        to="/register"
      >
        Register
      </NavLink>
    </li>
    <li
      className="li-nav"
    >
      <NavLink
        className="navLinks"
        to="/login"
      >
        Login
      </NavLink>
    </li>
  </ul>
</nav>
`;
